sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/Exchange.scala

```diff
@@ -58,6 +59,10 @@ case class ReusedExchangeExec(override val output: Seq[Attribute], child: Exchan
   override protected[sql] def doExecuteBroadcast[T](): broadcast.Broadcast[T] = {
     child.executeBroadcast()
   }
+
+  override def outputPartitioning: Partitioning = child.outputPartitioning
```

Comment:
`ReusedExchangeExec` can have distinct sets of output attribute ids. Shall we also update `outputPartitioning` and `outputOrdering`, if its output is different to `child.output`.